swift/xcode-autobuilder/XcodeProjectParser.cpp

```diff
@@ -217,41 +233,29 @@ static std::vector<fs::path> collectFiles(const std::string& workingDir) {
       continue;
     }
     if (p.extension() == ".xcodeproj" || p.extension() == ".xcworkspace") {
-      files.push_back(p);
+      structure.xcodeFiles.push_back(p);
```

Comment:
maybe as you are anyway creating this structure with fields for the different kinds of files, you can also split `.xcodeproj` and `.xcworkspace` in separate `xcodeProjects` and `.xcodeWorkspaces` fields of this structure, and avoid testing the extension again later?

Reply:
Yeah, I thought about this but decided not to as these two go hand in hand and it doesn't make much difference if we do it the other way.